pkg/ext/stores/tokens/tokens.go

```diff
@@ -375,15 +435,15 @@ func (t *SystemStore) Create(ctx context.Context, group schema.GroupResource, to
 		t.initialized = true
 	}
 
-	ensureNameOrGenerateName(token)
-	// we check the Name directly. because of the ensure... we know that
-	// GenerateName is not set. as it squashes the name in that case.
-	if token.Name != "" {
-		// reject creation of a token which already exists
-		currentSecret, err := t.secretCache.Get(TokenNamespace, token.Name)
-		if err == nil && currentSecret != nil {
-			return nil, apierrors.NewAlreadyExists(group, token.Name)
-		}
+	// enforce our choice of required generateName, and forbidden name.
+	if err := ensureNoNameAndGenerateName(token); err != nil {
+		return nil, err
+	}
+
+	// reject any attempt to specify status fields
+	emptyStatus := ext.TokenStatus{}
+	if token.Status != emptyStatus {
+		return nil, apierrors.NewBadRequest("Token is invalid: status is not empty")
```

Comment:
How does this work with `kubectl apply/edit`? Do users need to remove the status fields every time they are trying to make an unrelated change e.g. change the description?

Reply:
This code is in `create`. In update for edit I have
> 	// Keep the status of the resource unchanged, never store a token value, etc.
	// IOW changes to hash, value, etc. are all ignored without a peep.
	token.Status = currentToken.Status
	token.Status.Value = ""
	// Refresh time of last update to current.
	token.Status.LastUpdateTime = t.timer.Now()

Reply:
AFAIK core resources' status and immutable fields (like createTimestamp, UID etc) are ignored on create too. E.g.
```
k create deployment --image nginx mynginx
k get deployment mynginx -o yaml > mynginx2.yaml
vi mynginx2.yaml # change name and app label to mynginx2
k create -f mynginx2
```

Reply:
Ah. You are saying that standard resources simply ignore status on create. In that case it certainly makes sense for us to do the same.

Reply:
Done. By undoing the original addition. Rebased.